src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -239,166 +239,168 @@ protected void install(InstallConfig installConfig) {
         }
 
         inboundRateLimiter.clear();
+        config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe((what, newv) -> {
+            if (what.equals(WhatHappened.timestampUpdated)) {
+                return;
+            }
+            if (installConfig.configureSynchronizeConfig
+                    && (newv == null || newv.childOf(CONFIGURATION_SYNCHRONIZATION_TOPIC))) {
+                configureSynchronization();
```

Comment:
are any of these expensive in any way? I think I'd like to just run them all no matter what.

In fact, for the things which don't need to be event based (ie. they're just changing a config option like the max size) then I'd like them to not exist at all. I want the code to directly read from the config instead of reading a local field which is updated by the config (since that introduces an opportunity to be out of sync).

Reply:
+1 - the config parsing/validation has gotten a bit too defensive - it could be simplified - e.g. if anything under syncing changes - let the sync handler go and get whatever config it needs and restart. If rate limits change, tell the rate limiter to reload values